[PIM-42] 로그인 API 구현

Seyiul opened the pull request:
## 로그인 API 구현 <!-- 소셜 로그인 구현 -->

### 지라 티켓 번호
PIM-42

### PR 타입(하나 이상의 PR 타입을 선택해주세요)
- [x] 기능 추가
- [ ] 기능 삭제
- [ ] 버그 수정
- [ ] 의존성, 환경 변수, 빌드 관련 코드 업데이트

### 구현 내용
1. 이메일을 이메일 형식으로 입력하지않은 경우/ 비밀번호를 영문/숫자/특수문자의 조합으로 입력하지 않은 경우, 로그인이 제한될 수 있도록 구현
2. 모든 유효성 검사를 통과한 경우, workspace로 이동하도록 구현하였습니다.


### 리뷰 포인트
<!-- 오류 있는지 함께 확인해주세요 -->
- 이메일과 비밀번호를 형식에 맞게 입력한 경우, `/workspace` 로 이동하는 지 확인해주세요.
- 이메일과 비밀번호의 validation 검사가 제대로 이루어지는지 확인해주세요.

### TODO
<!-- 추가적으로 테스트 코드를 작성할 예정입니다. -->

SeonDahye commented on src/pages/authorization/sign-up/index.tsx:
```diff
@@ -0,0 +1,158 @@
+      </div>
+      <div css={RightWrapper}>
+        <div css={header}>
+          <div
+            css={css`
+              display: flex;
+              align-items: center;
+            `}
+          >
```
인라인 css 수정 부탁드려요 !! 💖

Seyiul replied:
해당 내용 수정 완료했습니다 확인해주세요 🥰🥰🥰

CHOSIYEON commented on src/pages/authorization/sign-up/styles.tsx:
```diff
@@ -0,0 +1,139 @@
+  margin-bottom: 30px;
+`;
+
+export const content = css`
+  display: flex;
+  flex-direction: column;
+  width: 100%;
+  justify-content: center;
+  align-items: center;
+`;
```
컨벤션 규칙 확인해주세용

Seyiul replied:
해당 내용 수정 완료했습니다 확인해주세요 🥰🥰🥰

CHOSIYEON reviewed (approved):
아주 잘하셨네요! 
코드별로 리뷰 남겨뒀습니당 ㅎㅎ 수고하셨어요~!

SeonDahye commented on src/pages/authorization/sign-up/styles.tsx:
```diff
@@ -0,0 +1,139 @@
+  line-height: 20px;
+  min-height: 28px;
+  border-radius: 2rem;
+  padding: 8px 40px;
+  border: 1px solid #eaebf6;
+  transition: all 0.1s ease 0s;
+  background: url('src/assets/authorization/login/padlock.png') no-repeat left;
+  background-size: 16px;
+  background-position: 10px 10px;
+
+  :focus {
+    border: 1.5px solid #e5a4fc;
+  }
+
+  ::placeholder {
+    color: #c0c3d0;
+  }
+`;
+export const nickname = css`
+  width: 50%;
+  color: rgb(36, 35, 42);
+  font-size: 16px;
+  line-height: 20px;
+  min-height: 28px;
+  border-radius: 2rem;
+  padding: 8px 40px;
+  border: 1px solid #eaebf6;
+  transition: all 0.1s ease 0s;
+  background: url('src/assets/authorization/login/profile.png') no-repeat left;
+  background-size: 16px;
+  background-position: 10px 10px;
+
+  :focus {
+    border: 1.5px solid #e5a4fc;
+  }
+
+  ::placeholder {
+    color: #c0c3d0;
+  }
+`;
```
idInput, passwordInput 등 네이밍을 구체적으로 하면 더 좋을 것 같아요!! 🙌

Seyiul replied:
해당 내용 수정 완료했습니다 확인해주세요 🥰🥰🥰

SeonDahye reviewed (approved):
화면이 아주 잘 구현되었어요 프로토타입과 똑같네요 ~~ 😊

Seyiul commented on src/pages/authorization/sign-up/modal.tsx:
```diff
@@ -0,0 +1,58 @@
+
+interface ModalDefaultType {
+  onClickToggleModal: () => void;
+}
```
해당 내용 수정 완료했습니다 확인해주세요 🥰🥰🥰

Seyiul commented on src/pages/authorization/sign-up/index.tsx:
```diff
@@ -0,0 +1,158 @@
+              src="src/assets/authorization/login/arrow-right.png"
+              css={css`
+                width: 18px;
+                margin-right: 8px;
```
해당 내용 수정 완료했습니다 확인해주세요 🥰🥰🥰

CHOSIYEON reviewed (approved):
수정 수고하셨습니다 !!!

Seyiul committed:
feat: 로그인 api 구현

Seyiul committed:
feat: form validation 추가

Seyiul committed:
feat: 로그인 validation 추가

- 이메일을 이메일 형식으로 입력하지않은 경우
- 비밀번호를 영문/숫자/특수문자의 조합으로 입력하지 않은 경우

로그인이 제한될 수 있도록 구현

Seyiul committed:
feat: utils > validation 추가

- utils 파일 생성 후, 로그인 및 회원가입에 필요한 validation을 저장

CHOSIYEON commented on src/pages/authorization/login/index.tsx:
```diff
@@ -22,17 +61,30 @@ function Login() {
+            <S.Warning>
+              <p hidden={emailValidation}>
+                영어/숫자/특수문자를 조합하여 8자리 이상 입력해주세요.
+              </p>
```
hidden 조건이 emailValidation이라 좀 오류가 있는 것 같아요! 
비밀번호 조건이 틀렸어도 이메일 조건이 맞으면 에러 메세지가 보이지 않네요 😂😂

Seyiul replied:
앗! 머쓱티비🤭🤭🤭🤭🤭🤭 비밀번호 유효성 검사로 수정했습니다! 다시 확인해주세요 헿

CHOSIYEON reviewed (changes requested):
validation 부분 확인 부탁드려욧

Seyiul committed:
feat: 회원가입 UI 구현

Seyiul committed:
Revert "feat: 회원가입 UI 구현"

This reverts commit 849cdb40d4702b3e54b11865d38c5018d0bbfed2.

Seyiul committed:
fix: 비밀번호 유효성검사 수정

CHOSIYEON reviewed (approved):
매우 잘 넘어가네용 수고하셨습니다~!

SeonDahye commented on src/pages/authorization/login/index.tsx:
```diff
@@ -22,17 +61,30 @@ function Login() {
+            <S.InputPwd type="password" onBlur={pwdInput} required></S.InputPwd>
+            <S.Warning>
+              <p hidden={pwdValidation}>
+                영어/숫자/특수문자를 조합하여 8자리 이상 입력해주세요.
```
대소문자 구분에 대한 안내도 함께 적으면 좋을 것 같아요!

SeonDahye reviewed (approved):
잘 작동되네요! 고생하셨어요! 💖💖
